### PR TITLE
Arrow functions

### PR DESCRIPTION
**Pull request by AJenbo**
Type: refactoring
Issue: https://github.com/phpmd/phpmd/wiki/3.0-Roadmap
Breaking change: no

### CODE REVIEW

File: src/main/php/PHPMD/Rule/CleanCode/IfStatementAssignment.php [hunk 1]
```diff
@@ -69,9 +69,7 @@ public function apply(AbstractNode $node): void
      */
     protected function getStatements(AbstractNode $node)
     {
-        return call_user_func_array('array_merge', array_map(function ($type) use ($node) {
-            return $node->findChildrenOfType($type);
-        }, $this->ifStatements));
+        return array_merge(...array_map(fn ($type) => $node->findChildrenOfType($type), $this->ifStatements));
```
Comment (kylekatarnls) [inline src/main/php/PHPMD/Rule/CleanCode/IfStatementAssignment.php:72]:
```suggestion
        return array_merge(...array_map(static fn ($type) => $node->findChildrenOfType($type), $this->ifStatements));
```

File: src/main/php/PHPMD/Rule/CleanCode/IfStatementAssignment.php [hunk 2]
```diff
@@ -82,9 +80,7 @@ protected function getStatements(AbstractNode $node)
      */
     protected function getExpressions(array $statements)
     {
-        return array_map(function (ASTNode $statement) {
-            return $statement->getFirstChildOfType('Expression');
-        }, $statements);
+        return array_map(fn (ASTNode $statement) => $statement->getFirstChildOfType('Expression'), $statements);
```
Comment (kylekatarnls) [inline src/main/php/PHPMD/Rule/CleanCode/IfStatementAssignment.php:83]:
```suggestion
        return array_map(
            static fn (ASTNode $statement) => $statement->getFirstChildOfType('Expression'),
            $statements,
        );
```

File: src/test/php/PHPMD/AbstractStaticTestCase.php [hunk 3]
```diff
@@ -97,9 +97,7 @@ protected static function createCodeResourceUriForTest()
      */
     protected static function getValuesAsArrays($values)
     {
-        return array_map(function ($value) {
-            return [$value];
-        }, $values);
+        return array_map(fn ($value) => [$value], $values);
```
Comment (kylekatarnls) [inline src/test/php/PHPMD/AbstractStaticTestCase.php:100]:
```suggestion
        return array_map(static fn ($value) => [$value], $values);
```

File: src/test/php/PHPMD/Rule/UnusedFormalParameterTest.php [hunk 4]
```diff
@@ -371,9 +371,7 @@ public function testRuleDoesNotApplyToImplementedInterfaceMethod()
      */
     public function testRuleDoesNotApplyToMagicMethod()
     {
-        $methods = array_filter($this->getClass()->getMethods(), function ($method) {
-            return $method->getName() === '__call';
-        });
+        $methods = array_filter($this->getClass()->getMethods(), fn ($method) => $method->getName() === '__call');
```
Comment (kylekatarnls) [inline src/test/php/PHPMD/Rule/UnusedFormalParameterTest.php:374]:
```suggestion
        $methods = array_filter(
            $this->getClass()->getMethods(),
            static fn ($method) => $method->getName() === '__call',
        );
```

File: src/test/php/PHPMD/Rule/UnusedFormalParameterTest.php [hunk 5]
```diff
@@ -529,9 +527,7 @@ public function testRuleDoesNotApplyToFormalParameterUsedAsParameterInStringComp
      */
     public function testRuleDoesNotApplyToPropertyPromotionParameters()
     {
-        $methods = array_filter($this->getClass()->getMethods(), function ($method) {
-            return $method->getImage() === '__construct';
-        });
+        $methods = array_filter($this->getClass()->getMethods(), fn ($method) => $method->getImage() === '__construct');
```
Comment (kylekatarnls) [inline src/test/php/PHPMD/Rule/UnusedFormalParameterTest.php:530]:
```suggestion
        $methods = array_filter(
            $this->getClass()->getMethods(),
            static fn ($method) => $method->getImage() === '__construct',
        );
```

**Review by kylekatarnls — APPROVED** [review]

**Review by kylekatarnls — APPROVED** [review]

**Comment by AJenbo** [comment]
Thanks, didn't get to applying this last night

**Pull request merged**